libcontainer/nsenter/nsenter_test.go

```diff
@@ -257,3 +238,23 @@ func initWaiter(t *testing.T, r io.Reader) {
 	}
 	t.Fatal("waiting for init preliminary setup:", err)
 }
+
+func reapChild(t *testing.T, parent *os.File) {
+	t.Helper()
+	decoder := json.NewDecoder(parent)
+	decoder.DisallowUnknownFields()
+	var pid struct {
+		Pid2 int `json:"stage2_pid"`
+		Pid1 int `json:"stage1_pid"`
+	}
+	if err := decoder.Decode(&pid); err != nil {
+		t.Fatal(err)
+	}
+	if pid.Pid1 == 0 || pid.Pid2 == 0 {
+		t.Fatal("got pids:", pid)
+	}
+
+	// Reap children.
+	_, _ = unix.Wait4(pid.Pid1, nil, 0, nil)
```

Comment:
Would reaping still be needed if either Pid1 or Pid2 is `!= 0`?

Reply:
In this case the test will fail, the test process will end, and these processes will be reparented to pid 1 (init) who'll reap them. So, technically, this reaping is not needed at all (until one day in a very distant future we'll have so many cases that the number of zombie processes will hit any of the PID limits, and then we'll have all sorts of mysterious failures).

Yet it makes sense to reap before we check for `0` (as we do it on a "best effort" basis anyway).

Reply:
done